### ISSUE TITLE
No support for open ssh keys?

### ISSUE THREAD

**Issue by charpov** [sbt/sbt#5589]
## steps

I am able to publish to a private repository with this configuration:

```
  publishTo := {
    val key = file(Path.userHome.absolutePath + "/.ssh/id_rsa")
    Some(Resolver.ssh("...", "...", 22, "...")
      as("...", key) withPublishPermissions "0644")
  }
```
but not if I try to use an open ssh key:
```
    val key = file(Path.userHome.absolutePath + "/.ssh/id_ed25519")
```
## problem
Error message is:
```
java.io.IOException: invalid privatekey
```
## expectation

My expectation is that the newer key format would be supported.

**Comment by eed3si9n** [comment]
I think this might be related what I reported to JSch? - "JSchException: invalid privatekey" on OpenSSH 7.8 and above https://sourceforge.net/p/jsch/bugs/129/

**Comment by charpov** [comment]
> I think this might be related what I reported to JSch? - "JSchException: invalid privatekey" on OpenSSH 7.8 and above https://sourceforge.net/p/jsch/bugs/129/

Probably.  I was wondering if it's an sbt thing or a Java thing.

**Comment by magnolia-k** [comment]
Giter8 seems to be able to recognize OpenSSH format keys by eliminating the dependency on jsch with the following changes in the latest 0.14.0 version.

https://github.com/foundweekends/giter8/commit/c51d1711815db44cc4458b81b286ad26c9132598

When I prepare an OpenSSH key and run Giter8 directly, I get an error with 0.13.1 (the version used by sbt), but not with 0.14.0.

It seems that the version of Giter8 that sbt depends on should be changed to 0.14.0.

**Comment by magnolia-k** [comment]
@charpov 

fixed in sbt 1.7.2

https://github.com/sbt/sbt/releases/tag/v1.7.2

**Comment by eed3si9n** [comment]
Closing this as fixed.
